gubernator/github/classifier.py

```diff
@@ -21,6 +21,33 @@
 XREF_RE = re.compile(r'k8s-gubernator.appspot.com/build(/[^])\s]+/\d+)')
 
 
+class Deduper(object):
+    ''' Arena intern() equivalent.
+
+    Tries to reduce memory usage by making equivalent strings point at the same object.
+    This reduces memory usage for large, repetitive JSON structures by >2x. '''
+
+    def __init__(self):
+        self.strings = {}
+
+    def dedup(self, obj):
+        if isinstance(obj, basestring):
+            return self.strings.setdefault(obj, obj)
+        elif isinstance(obj, dict):
+            for k in obj.keys():
+                obj[k] = self.dedup(obj[k])
+            return obj
+        elif isinstance(obj, tuple):
+            for x in obj:
+                self.dedup(x)
```

Comment:
It doesn't look like this dedups the strings in the tuple?

Maybe something like: return tuple(self.dedup(x) for x in obj)?